app/src/main/java/com/babylon/wallet/android/domain/usecases/CreateAccountWithBabylonDeviceFactorSourceUseCase.kt

```diff
@@ -0,0 +1,81 @@
+import rdx.works.profile.derivation.model.NetworkId
+import rdx.works.profile.di.coroutines.DefaultDispatcher
+import rdx.works.profile.domain.EnsureBabylonFactorSourceExistUseCase
+import javax.inject.Inject
+
+class CreateAccountWithBabylonDeviceFactorSourceUseCase @Inject constructor(
+    private val mnemonicRepository: MnemonicRepository,
+    private val ensureBabylonFactorSourceExistUseCase: EnsureBabylonFactorSourceExistUseCase,
+    private val profileRepository: ProfileRepository,
+    private val resolveAccountsLedgerStateUseCase: ResolveAccountsLedgerStateUseCase,
+    @DefaultDispatcher private val defaultDispatcher: CoroutineDispatcher
+) {
+
+    private val _interactionState = MutableStateFlow<InteractionState?>(null)
+    val interactionState: Flow<InteractionState?> = _interactionState.asSharedFlow()
+
+    suspend operator fun invoke(
+        displayName: String,
+        networkID: NetworkId? = null
+    ): Network.Account {
+        return withContext(defaultDispatcher) {
+            val profile = ensureBabylonFactorSourceExistUseCase()
+            val factorSource = profile.mainBabylonFactorSource()
+                ?: error("Babylon factor source is not present")
+            _interactionState.update { InteractionState.Device.DerivingAccounts(factorSource) }
+
+            // Construct new account
+            val networkId = networkID ?: profile.currentNetwork?.knownNetworkId ?: Radix.Gateway.default.network.networkId()
+            val nextAccountIndex = profile.nextAccountIndex(DerivationPathScheme.CAP_26, networkId, factorSource.id)
+            val nextAppearanceId = profile.nextAppearanceId(networkId)
+            val mnemonicWithPassphrase = requireNotNull(mnemonicRepository.readMnemonic(factorSource.id).getOrNull())
+            val newAccount = initAccountWithBabylonDeviceFactorSource(
+                entityIndex = nextAccountIndex,
+                displayName = displayName,
+                mnemonicWithPassphrase = mnemonicWithPassphrase,
+                deviceFactorSource = factorSource,
+                networkId = networkId,
+                appearanceID = nextAppearanceId
+            )
+            // Add account to the profile
+            val resolveResult = resolveAccountsLedgerStateUseCase.invoke(listOf(newAccount))
+            // Add account to the profile
+            val updatedProfile = if (resolveResult.isSuccess) {
+                profile.addAccounts(
+                    accounts = listOf(resolveResult.getOrThrow().first().account),
+                    onNetwork = networkId
+                )
+            } else {
+                profile.addAccounts(
+                    accounts = listOf(newAccount),
+                    onNetwork = networkId
+                )
+            }
+            // Save updated profile
+            profileRepository.saveProfile(updatedProfile)
+            _interactionState.update { null }
+            // Return new account
+            newAccount
```

Comment:
If account with ledger details is received, don't we need to return it rather than returning the new account?

What if 
```
val resolvedAccount = if (resolveResult.isSuccess) {
    resolveResult.getOrThrow().first().account
else {
    newAccount
}
profile.addAccounts(
    accounts = listOf(resolvedAccount),
    onNetwork = networkId
)

return resolvedAccount
```